src/main/scala/EmrSparkPlugin.scala

```diff
@@ -261,37 +293,41 @@ object EmrSparkPlugin extends AutoPlugin {
     }
   }
 
-  def submitJob(
-    settings: Settings,
-    mainClass: String,
-    args: Seq[String],
-    jar: File
-  )(implicit log: Logger) = {
-    log.info("Uploading the jar.")
-
+  def uploadJarToS3(s3Location: String, jarFile: String)(implicit log: Logger) = {
+    log.info("Uploading jar to S3 path " + s3Location)
     val s3 = AmazonS3ClientBuilder.defaultClient()
-    val jarUrl = new S3Url(settings.s3JarFolder) / jar.getName
-
-    s3.putObject(jarUrl.bucket, jarUrl.key, jar)
+    val jarUrl = new S3Url(s3Location)
```

Comment:
Any reason not to keep it as this?
```
val jarUrl = new S3Url(settings.s3JarFolder) / jarFile.getName
```
`S3Url()` will help handling the trailing slash problem. Since we are not sure if `s3JarFolder` will be `s3://abc` or `s3://abc/`, directly concatenating it like `settings.s3JarFolder + jar.getName` may give error?